### PR TITLE
(#3501) Remove old install migration code

### LINKED ISSUE #3501: Chocolatey folder migration logic should be removed from `chocolateysetup.psm1` file
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my request.

### Is Your Feature Request Related To A Problem? Please describe.

No.  There is no specific problem associated with this task, it is more around tidying up of code that is no longer required.

### Describe The Solution. Why is it needed?

Within the chocolateysetup.psm1 file, there are a number of functions that were specifically created to aid with the migration from the old Chocolatey CLI installation folder `C:\Chocolatey` to the new one `C:\ProgramData\Chocolatey`.

This code is no longer required, and can be removed.

### Additional Context

The migration code was added to the chocolateysetup.psm1 file in version 0.9.9, which was shipped in March 2015.  Chocolatey CLI is now on its second major version, and the upgrade steps to get to 2.x first involve getting to 1.4.0.  As a result, this migration code is no longer required.

### Related Issues

N/A

### PR DESCRIPTION
**Pull request by vexx32**
## Description Of Changes
- Remove code relating to migration from a `C:\chocolatey` install folder from the `chocolateysetup.psm1` module used for installing Chocolatey.

## Motivation and Context
This migration path hasnt been supported or used since pre-1.0 versions of Chocolatey and is dead code.

Removing this now to simplify the install process somewhat and get the old code out of the way as it's not used at all.

## Testing

1. Run `build.ps1` to generate a package
2. Copy the generated nupkg to a fresh VM and take a snapshot
3. Install Chocolatey using the following PowerShell code from an admin shell, substituting the nupkg path for the nupkg you copied over:
   ```powershell
   ${function:Install-Choco} = [scriptblock]::Create((Invoke-WebRequest https://community.chocolatey.org/install.ps1))
   Install-Choco -ChocolateyDownloadUrl ./path/to/chocolatey.nupkg
   ```
4. Ensure the installation completes successfully
5. Downgrade choco to 2.3.0:  `choco upgrade chocolatey --allow-downgrade --version 2.3.0 --allow-unofficial`
6. Upgrade choco once again to the built nupkg to ensure the upgrade works appropriately: `choco upgrade chocolatey --source ./path/to/nupkg/folder --pre`
7. Ensure the upgrade completes successfully
8. Restore from snapshot
9. Set `$env:ChocolateyInstall = 'C:\Users\YOUR_USERNAME\chocolatey'`
10. Repeat from step 3 _in a non-admin shell_ to verify things work with a non-admin install as well

### Operating Systems Testing

Windows 10

## Change Types Made

* [ ] Bug fix (non-breaking change).
* [x] Feature / Enhancement (non-breaking change).
* [ ] Breaking change (fix or feature that could cause existing functionality to change).
* [ ] Documentation changes.
* [ ] PowerShell code changes.

## Change Checklist

* [ ] Requires a change to the documentation.
* [ ] Documentation has been updated.
* [ ] Tests to cover my changes, have been added.
* [x] All new and existing tests passed?
* [x] PowerShell code changes: PowerShell v3 compatibility checked?

## Related Issue

Fixes #3501

### CODE REVIEW

**Review by corbob — COMMENTED** [review]
Code wise this looks good. I still need to give it a test locally.

File: nuspec/chocolatey/chocolatey/tools/chocolateysetup.psm1 [hunk 1]
```diff
@@ -2,7 +2,6 @@ $thisScriptFolder = (Split-Path -Parent $MyInvocation.MyCommand.Definition)
 $chocoInstallVariableName = "ChocolateyInstall"
 $sysDrive = $env:SystemDrive
```
Comment (gep13) [inline nuspec/chocolatey/chocolatey/tools/chocolateysetup.psm1:3]: This can be removed as well, right?

**Review by gep13 — CHANGES_REQUESTED** [review]

File: nuspec/chocolatey/chocolatey/tools/chocolateysetup.psm1 (hunk 1 above)
Comment (vexx32) [reply nuspec/chocolatey/chocolatey/tools/chocolateysetup.psm1]: Yep, looks like, good catch! Removed :3

File: nuspec/chocolatey/chocolatey/tools/chocolateysetup.psm1 [hunk 2]
```diff
@@ -142,7 +140,7 @@ function Initialize-Chocolatey {
 
     # if we have an already environment variable path, use it.
     $alreadyInitializedNugetPath = Get-ChocolateyInstallFolder
-    if ($alreadyInitializedNugetPath -and $alreadyInitializedNugetPath -ne $chocolateyPath -and ($allowInsecureRootInstall -or $alreadyInitializedNugetPath -ne $defaultChocolateyPathOld)) {
+    if ($alreadyInitializedNugetPath -and $alreadyInitializedNugetPath -ne $chocolateyPath -and $allowInsecureRootInstall) {
```
[the remark this answers is not in the document]
Comment (vexx32) [reply nuspec/chocolatey/chocolatey/tools/chocolateysetup.psm1]: I think I have this sorted for now :3

**Review by corbob — APPROVED** [review]
Looks good to me.

**Pull request merged**